### PR TITLE
feat(ColorPicker): introduce color picker component

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #(issue_number)

### CODE REVIEW

File: src/components/color-picker/RuiColorHue.vue [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  modelValue: number;
+}>();
+
+const emit = defineEmits<{
+  (e: 'update:model-value', hue: number): void;
+}>();
+
+const barElement: Ref<HTMLElement | null> = ref(null);
+
+const modelValue = useVModel(props, 'modelValue', emit);
```
Comment (kelsos) [inline src/components/color-picker/RuiColorHue.vue:15]: does this send the proper event? I can see in the other place we had to specify the kebab-case event specifically.

Comment (lukicenturi) [reply src/components/color-picker/RuiColorHue.vue:15]: yes it works, at least in vue 3, I think `@update:model-value` just the same with `@update:modelValue`.
If I emit `update:modelValue` or `update:model-value`, both `@update:modelValue` and `@update:model-value` will be able to catch it.

File: src/components/color-picker/RuiColorInput.vue [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  modelValue: Color;
+}>();
+
+const emit = defineEmits<{
+  (e: 'update:model-value', color: Color): void;
+}>();
+
+const modelValue = useVModel(props, 'modelValue', emit, {
+  eventName: 'update:model-value',
+});
```
Comment (kelsos) [inline src/components/color-picker/RuiColorInput.vue:19]: this is what triggered the other comment @lukicenturi do we still need the eventName here? or can we just remove it?

Comment (lukicenturi) [reply src/components/color-picker/RuiColorInput.vue]: yeah I will remove it

**Review by kelsos — APPROVED** [review]
lgtm, thanks @lukicenturi

**Pull request merged**